perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/facilitiestabs/FacilitySettingsTabItem.java

```diff
@@ -136,19 +160,43 @@ public Widget draw() {
 
 		// services listbox
 		final ListBoxWithObjects<Service> servList = new ListBoxWithObjects<Service>();
+		if (switchServicesChb.getValue() == true) {
+			lastServiceId = 1;
```

Comment:
Please do not set "lastServiceId" to 1 since it supposed to be ID of a real Service object (not its index in a listbox). It worked previously with 0, since by DB sequence definition we start counting services with 1. Like this you couldn't automatically select service with id=1 and "all" option would be selected instead.

Create own global variable where you can store last selected index and if relevant, use it instead of last service id (e.g. when service id=0, read index which must be either 0 (Not selected) or 1 (All - only if present in listbox) ). Thank you.